tensorflow/core/platform/s3/s3_file_system.cc

```diff
@@ -262,8 +262,14 @@ class S3WritableFile : public WritableFile {
     outfile_->clear();
     outfile_->seekp(offset);
     if (!putObjectOutcome.IsSuccess()) {
-      return errors::Unknown(putObjectOutcome.GetError().GetExceptionName(),
-                             ": ", putObjectOutcome.GetError().GetMessage());
+
+      auto error = putObjectOutcome.GetError();
+      if (error.GetResponseCode() == Aws::Http::HttpResponseCode::FORBIDDEN) {
+        return errors::FailedPrecondition("AWS Credentials have not been set properly. "
+                                          "Unable to access the specified S3 location");
+      } else {
+        return errors::Unknown(error.GetExceptionName(), ": ", error.GetMessage());
+      }
```

Comment:
Can we extract this (and the copies below) to a separate `CheckForCredentialError` (or some other better suited name)?

Reply:
On it

Reply:
Done